hschain-PoW/HSChain/PoW/Types.hs

```diff
@@ -176,7 +176,7 @@ class BlockData b => Mineable b where
   --
   -- One can use second part in a parameter search process. It is a
   -- proxy to how many hash attempts are done during search.
-  adjustPuzzle :: MonadIO m => Block b -> m (Maybe (Block b), Target)
+  adjustPuzzle :: MonadIO m => Block b -> m (Maybe (Block b))
```

Comment:
Second part is important - it can be used in search of better parameters. We can fail to obtain a block yet we can find something with a not valid target.

This target value can be used to guide search for better solution search parameters.

It is valid for our PoW (parameters of SLS solver) and also valid for Monero/RandomX for one example (JIT parameters).

I would better keep that.

Reply:
The reason it is not used is lack of time and/or willpower. I had a parameters search process based on NES prototyped but had no time to port it here.

Reply:
I see. 

There's another thing but it could be addressed later I think. There's more than one algorithm for mining: different SAT solvers/different ASIC/etc. So type class is poor interface. There's no (sane) way to encode multiple mining algorithms for single block type